demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -1018,6 +1034,15 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
                                                    ( uint8_t * ) pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS,
                                                    &xProvisionedState.xPrivateKey,
                                                    &xProvisionedState.xPublicKey );
+
+            /* Get the bytes of the new public key. */
+            if( CKR_OK == xResult )
```

Comment:
It seems like xGeneratedPrivateKey should be set to true somewhere in here.

Reply:
Good catch. Fixed.